content/docs/architecture/resources/environment.md

```diff
@@ -1,66 +1,87 @@
+  template_mapping:
+    example-do-vm-template:
+      image: ubuntu-20-04-x64
+      size: s-1vcpu-512mb-10gb
+      ssh_username: root
+  ws_endpoint: shell.{domain}.com
 ```
 
 ## Configuration
 
 ### `count_capacity`
+This field is a `map[string]int` and is designed to inform HobbyFarm about the capacity of the environment for a particular VM template. Each key in this map is the name of the VM template, and the value is the number of total VMs of that type that the environment can support.
 
-This field is a `map[string]int` designed to inform HobbyFarm about the capacity of the environment for a particular VM template. Each key in this map is the name of the VM template, and the value is the number of total VMs of that type that the environment can support. 
-
-For example:
-```yaml
-...
-count_capacity:
-    example-template: 100
-```
+This map should be set according to what resources an environment has. Take care to account for multiple templates and totals that will affect the amount of instances, cpu, memory, etc. that are consumed in the provider.
 
-The above example indicates that at any given point in time, the environment can support up to 100 total `example-template` VMs.
-
-This map should be set according to what resources an environment has. Take care to account for multiple templates and totals that will affect the amount of instances, cpu, memory, etc. that are consumed in the provider. 
+> **NOTE:** In the example manifest shown above, the environment can support 10 total VMs of the `example-do-vm-template` type.
 
 ### `display_name`
-
 Display name for the environment.
 
 ### `environment_specifics`
+This field is a `map[string]string` designed to inform HobbyFarm about specific values for the environment. Entries in this map are utilized by VM provisioners for purposes.
+
+The keys defined here are specific to the provisioner in use for the environment, such as DigitalOcean or AWS. Each provisioner may have specific values that are required.
 
-This field is a `map[string]string` designed to inform HobbyFarm about specific values for the environment. Entries in this map are utilized by VM provisioners for purposes such as security group ID, VPC, etc. The keys defined here are specific to the provisioner in use for the environment. Each provisioner may have specific values that are required.
+> **NOTE:** In the example manifest shown above, the DigitalOcean provisioner requires a `region` and `token-secret` to be defined.
 
 ### `provider`
+This field is used to determine what provisioner should be utilized for an environment and **_must_** be used in conjunction with the `hobbyfarm.io/provisioner: external` annotation to set the external provisioner. See the [example manifest above](#environment-manifest-example) for a demonstration of this annotation.
 
-This field is used to determine what provisioner should be utilized for an environment. 
+> **NOTE:** Current available options are `digitalocean` and `aws`.
```

Comment:
with the terraform module enabled there are unlimited available provisioners.
however the field is ignored and the environment_specific option "module" is being used

Reply:
@jggoebel I specifically didn't mention the Terraform module because it has been deprecated, and was only using the current provisioners as the use-case. 

Should I include Terraform despite its deprecation?

Reply:
For now, yes, as we do not have a slated date for terraform removal.

Reply:
@ebauman OK, I'll make the necessary updates.